tuna/connection.py

```diff
@@ -241,36 +263,43 @@ def ssh_connect(self, abort=None):
     self.ssh = paramiko.SSHClient()
     self.ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
     for ssh_idx in range(NUM_SSH_RETRIES):
-      if abort is not None and chk_abort_file(self.id, self.logger):
-        self.logger.warning('Machine %s aborted ssh connection', self.id)
+      if abort is not None and chk_abort_file(id, self.logger):
+        self.logger.warning('Machine %s aborted ssh connection', id)
         return False
 
       try:
-        self.ssh.connect(self.hostname,
-                         username=self.user,
-                         password=self.password,
-                         port=self.port,
-                         timeout=SSH_TIMEOUT,
-                         allow_agent=False)
+        self.ssh.connect(  #type: ignore
+            self.hostname,  #type: ignore
+            username=self.user,  #type: ignore
+            password=self.password,  #type: ignore
+            port=self.port,  #type: ignore
+            timeout=SSH_TIMEOUT,
+            allow_agent=False)
       except paramiko.ssh_exception.BadHostKeyException:
         self.ssh = None
         self.logger.error('Bad host exception which connecting to host: %s',
-                          self.hostname)
+                          self.hostname)  #type: ignore
       except (paramiko.ssh_exception.SSHException, socket.error):
-        retry_interval = randrange(SSH_TIMEOUT)
+        retry_interval = randrange((int(SSH_TIMEOUT)))
         self.logger.warning(
             'Attempt %s to connect to machine %s (%s p%s) via ssh failed, sleeping for %s seconds',
-            ssh_idx, self.id, self.hostname, self.port, retry_interval)
+            ssh_idx,
+            id,
+            self.hostname,  #type: ignore
+            self.port,  #type: ignore
+            retry_interval)  #type: ignore
```

Comment:
let's clean up these ignores.

Reply:
all local variables are initialized in __init__ , this would enable the self key to recognize class variables. mypy thows error , if class memeber variables are not initialized.